gobblin-utility/src/test/java/gobblin/util/SchedulerUtilsTest.java

```diff
@@ -90,6 +91,13 @@ public void setUp()
     jobProps4.setProperty("k5", "b5");
     // test-job-conf-dir/test2/test21.pull
     jobProps4.store(new FileWriter(new File(subDir2, "test21.pull")), "");
+
+    Properties jobPropsWithInclude = new Properties();
+    jobPropsWithInclude.setProperty("include", jobsProps3File.getPath());
+    jobPropsWithInclude.setProperty("k8", "newValue8");
+    // test-job-conf-dir/test2/test22.pull
+    jobProps3.store(new FileWriter(new File(subDir2, "test22.pull")), "");
```

Comment:
Now, that I have the test being run. I found some aspect of Apache commons PropertiesConfiguration to be rather unintuitive and counter typical behavior of configuration files.

1) Repeated setting of the same key appends to the value rather then overwriting it:

a=1
a=2

Then, the effective value of "a" will be "1,2" rather than "2". This is in fact documented:
"If a key is used more than once, the values are appended like if they were on the same line separated with commas."

This is not how configuration files typically work. This is also not how java.util.Properties works.

The above also causes that configuration includes to behave non-intuitively:

test111.pull:
k8=a8

test21.pull:
include=test111.pull
k8=newValue

The effective value of k8 is "a8,newValue" rather than "newValue". Certainly, not what I would expect.


2) It seems like variable expansion (interpolation in their lingo) is done early rather than late.
In my example:

test111.pull:
k8=a8
k9=${k8}

test21.pull:
include=test111.pull
k8=newValue

These are the asserts that pass:
    Assert.assertEquals(jobPropsWithInclude.getProperty("k8"), "a8,newValue8");
    Assert.assertEquals(jobPropsWithInclude.getProperty("k9"), "a8");

k9 is expanded using the original value of k8, not the effective value. Again, this is not how I would generally expect for a configuration to work. 

I think we should check if there other options for property expansion or have our own PropertiesConfiguration implementation that is more intuitive.